src/aiida/orm/nodes/data/structure.py

```diff
@@ -1762,21 +1762,25 @@ def _get_object_pymatgen_structure(self, **kwargs):
         :return: a pymatgen Structure object corresponding to this
           :py:class:`StructureData <aiida.orm.nodes.data.structure.StructureData>`
           object
-        :raise ValueError: if periodic boundary conditions does not hold
-          in at least one dimension of real space; if there are partial occupancies
-          together with spins (defined by kind names ending with '1' or '2').
+        :raise ValueError: if the cell is not set (i.e. is the default one);
+          if there are partial occupancies together with spins
+          (defined by kind names ending with '1' or '2').
 
         .. note:: Requires the pymatgen module (version >= 3.0.13, usage
             of earlier versions may cause errors)
         """
+        import numpy as np
+        from pymatgen.core.lattice import Lattice
         from pymatgen.core.structure import Structure
 
-        if self.pbc != (True, True, True):
-            raise ValueError('Periodic boundary conditions must apply in all three dimensions of real space')
-
         species = []
         additional_kwargs = {}
 
+        if np.isclose(self.cell, _DEFAULT_CELL, 1e-10).all():
```

Comment:
If I understand correctly, this type of check is necessary instead of just checking for `None`, because `cell` always gets set? If so, might make sense to put a quick comment here because it looks like a weird way for checking whether a cell is defined at first glance

Reply:
It is correct, cell is set to the "default cell", which is a singular matrix. The latter is actually a better approach, and it is already done by pymatgen - so maybe a try/except around the return would be better?

For example:
```python
try:
  return Structure(lattice, ...)
except ValueError as err:
  raise ValueError('Singular matrix detected. Probably the cell was not set?') from err
```

Reply:
yep, if that's what pymatgen does, makes sense to use the approach of catching the exception instead 👍

Reply:
I am not familiar with pymatgen but how does it handle molecules in gas phase where pbc=(False, False, False)? 
The current check will prevent case to work, right? Or am I misunderstanding something?

Reply:
Ah, I see, this function only handle periodic structure, there are different functions that handle molecules.